### ISSUE TITLE
Unable to resolve service for type 'MyApp.Data.MyDBContext' while attempting to activate 'MyApp.Services.MyService'.at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites

### ISSUE THREAD

**Issue by faceoffers28** [dotnet/runtime#66148]
I'm trying to inject a Service from another Project into my Asp.Net 5.0 Function. The Service I'm injecting uses the DBContext. Every time I try to inject the DBContext I get the following error.

Unable to resolve service for type 'MyApp.Data.MyDBContext' while attempting to activate 'MyApp.Services.MyService'.at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites

I've tried so many ways. Based on everything I've read, this should work.

```
var host = new HostBuilder()
                .ConfigureFunctionsWorkerDefaults()
                .ConfigureServices(s =>
                {
s.AddLogging();
s.AddDbContext<MyDBContext>();
s.AddTransient<IMyService, MyService>();
})
.Build();
```

Is this a bug or am I calling it wrong?

**Comment by eerhardt** [comment]
What does `MyDBContext`,  `IMyService` and `MyService` look like?

**Comment by faceoffers28** [comment]
@eerhardt Thanks for the quick response.

The DBContext looks like this.

```
public class MyDBContext : DbContext
    {
        public MyDBContext(DbContextOptions<MyDBContext> options)
            : base(options)
        {
        }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlServer(@"CONNECTIONSTRING",
                    x => x.UseNetTopologySuite())
                .UseLoggerFactory(MyLoggerFactory);

            base.OnConfiguring(optionsBuilder);
        }

        public static readonly ILoggerFactory MyLoggerFactory = LoggerFactory.Create(builder =>
        {
            builder
                .AddFilter((category, level) =>
                    category == DbLoggerCategory.Database.Command.Name
                    && level == LogLevel.Information)
                .AddConsole();
        });

        public DbSet<MyEntity> MyEntities { get; set; }
        }

```
MyService looks like this.

```
public class MyService : IMyService
    {
        private readonly MyDBContext _context;

        public MyService(MyDBContext context)
        {
            _context = context;
        }

        public async Task...

        public async Task...

}
```

IMyService looks like this.

```
public interface IMyService
    {
        Task...
         
        Task...

    }

```

**Comment by eerhardt** [comment]
Hmm, the following works just fine for me:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.2" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.1" />
  </ItemGroup>
</Project>
```

```C#
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;

var host = new HostBuilder()
                //.ConfigureFunctionsWorkerDefaults()
                .ConfigureServices(s =>
                {
                    s.AddLogging();
                    s.AddDbContext<MyDBContext>();
                    s.AddTransient<IMyService, MyService>();
                })
.Build();

var s = host.Services.GetRequiredService<IMyService>();
Console.WriteLine(s.Context);

public class MyDBContext : DbContext
{
    public MyDBContext(DbContextOptions<MyDBContext> options)
        : base(options)
    {
    }

    //protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    //{
    //    optionsBuilder.UseSqlServer(@"CONNECTIONSTRING",
    //            x => x.UseNetTopologySuite())
    //        .UseLoggerFactory(MyLoggerFactory);

    //    base.OnConfiguring(optionsBuilder);
    //}

    public static readonly ILoggerFactory MyLoggerFactory = LoggerFactory.Create(builder =>
    {
        builder
            .AddFilter((category, level) =>
                category == DbLoggerCategory.Database.Command.Name
                && level == LogLevel.Information)
            .AddConsole();
    });

    public DbSet<MyEntity> MyEntities { get; set; }
}

public class MyEntity
{
}

public class MyService : IMyService
{
    private readonly MyDBContext _context;

    public MyService(MyDBContext context)
    {
        _context = context;
    }

    public MyDBContext Context => _context;
}

public interface IMyService
{
    MyDBContext Context { get; }

}
```

I'm not sure what `ConfigureFunctionsWorkerDefaults` is. Is that causing problems here?

**Comment by faceoffers28** [comment]
@eerhardt I've built an Asp.Net 5.0 Function. Looks like you are on 6. I'm testing in Azure. The error message I'm seeing occurs in Azure when the Function BLob Trigger fires.

**Comment by eerhardt** [comment]
It also works with:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.0" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
  </ItemGroup>
</Project>
```

Any chance you could provide a project that reproduces the problem? It is hard for me to guess without being able to reproduce it myself.

**Comment by faceoffers28** [comment]
@eerhardt Here are my packages from the Function.

```
<ItemGroup>
    <PackageReference Include="CsvHelper" Version="27.2.1" />
    <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.20.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.6.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs" Version="5.0.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.3.0" OutputItemType="Analyzer" />
    <PackageReference Include="Microsoft.Extensions.Azure" Version="1.1.1" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.2" />
  </ItemGroup>


```

**Comment by eerhardt** [comment]
@faceoffers28 - please see https://github.com/dotnet/runtime/blob/main/CONTRIBUTING.md#writing-a-good-bug-report, specifically "Why are Minimal Reproductions Important?". 

Unfortunately I am not able to reproduce the issue you are reporting. Without a minimal reproduction, I don't believe we will be able to help much here.

**Comment by faceoffers28** [comment]
@eerhardt I finally got passed this issue with the LocationService. I was Publishing the Function using the zip method. Anyway, the Publish to folder method, in Visual Studio, is not working correctly, I wasted hours and hours on this. The next thing is that if you comment out .ConfigureFunctionsWorkerDefaults() you get this... "Exception while executing function: Functions.ProcessFile Did not find any initialized language workers". The Function appears to be working now. I was finally able to get the normal web method of Publishing to work.
